scrapper.py

```diff
@@ -21,77 +39,245 @@ class IncorrectNumberOfArticlesError(Exception):
     """
 
 
-class UnknownConfigError(Exception):
-    """
-    Most general error
+class IncorrectStatusCode(Exception):
+    """"
+    Custom error
     """
 
 
+headers = {
+        'user-agent':
+            'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 '
+            '(KHTML, like Gecko) Chrome/88.0.4324.190 Safari/537.36'}
+
+
 class Crawler:
     """
     Crawler implementation
     """
-    def __init__(self, seed_urls: list, max_articles: int):
-        pass
+
+    def __init__(self, seed_urls: list, max_articles: int, max_articles_per_seed: int):
+        self.seed_urls = seed_urls
+        self.max_articles_per_seed = max_articles_per_seed
+        self.max_articles = max_articles
+        self.urls = []
 
     @staticmethod
     def _extract_url(article_bs):
-        pass
+        page_links = []
+        for tag_a_content in article_bs.find_all('a'):
+            get_url = tag_a_content.get('href')
+            if get_url:
+                re_url = re.findall(r'(/\w+/(\w+[-])+\w+\.html)', get_url)
+                if re_url:
+                    if get_url == re_url[0][0] and get_url not in page_links:
+                        page_links.append('http://ks-yanao.ru' + get_url)
+        return page_links
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for seed_url in self.seed_urls:
+            try:
+                response = requests.get(seed_url, headers=headers)
+                sleep(randint(2, 6))
+                if response.status_code:
+                    main_page_soup = BeautifulSoup(response.content, features='lxml')
+                else:
+                    raise IncorrectStatusCode
+            except IncorrectStatusCode:
+                continue
+            else:
+                found_links = self._extract_url(main_page_soup)
+                if len(found_links) < self.max_articles_per_seed:
+                    self.urls.extend(found_links)
+                else:
+                    self.urls.extend(found_links[:self.max_articles_per_seed])
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
+
+
+class CrawlerRecursive(Crawler):
+    """
+    Recursive Crawler
+    """
+
+    def find_articles(self):
+        """
+        Finds articles
+        """
+        if os.path.exists(os.path.join(PROJECT_ROOT, 'saved_urls.json')):
+            with open(os.path.join(PROJECT_ROOT, 'saved_urls.json'), encoding='utf-8') as file:
```

Comment:
it is initialization logic, is not it? then there should be a better place for initialization